tools/database/src/aggressive_trimming.rs

```diff
@@ -0,0 +1,194 @@
+                non_inlined_keys.len(),
+                non_inlined_keys.iter().map(|(size, _)| size).sum::<usize>()
+            );
+        }
+
+        {
+            let non_inlined_keys = non_inlined_keys.clone();
+            StoreParallelIterator::for_each_in_range(
+                store.clone(),
+                DBCol::FlatStateChanges,
+                Vec::new(),
+                Vec::new(),
+                6,
+                move |key: &[u8], value| {
+                    let key = KeyForFlatStateDelta::try_from_slice(key).unwrap();
+                    if key.shard_uid.version != sharding_version {
+                        return;
+                    }
+                    let changes = FlatStateChanges::try_from_slice(value).unwrap();
+                    for (_, value) in changes.0 {
+                        if let Some(FlatStateValue::Ref(r)) = value {
+                            let mut keys = non_inlined_keys.lock().unwrap();
+                            keys.push((
+                                r.length as usize,
+                                encode_flat_state_db_key(key.shard_uid, &r.hash.0),
+                            ));
+                            if keys.len() % 10000 == 0 {
+                                println!("Found {} keys to read", keys.len());
+                            }
+                        }
+                    }
+                },
+                true,
+            );
+        }
+
+        {
+            let non_inlined_keys = non_inlined_keys.lock().unwrap();
+            println!(
+                "Found {} non-inlined keys from FlatState and FlatStateChanges, total size {}",
+                non_inlined_keys.len(),
+                non_inlined_keys.iter().map(|(size, _)| size).sum::<usize>()
+            );
+        }
+
+        // Now read the non-inlined keys from the State column.
+        let non_inlined_entries = Arc::new(Mutex::new(BTreeMap::<Vec<u8>, Vec<u8>>::new()));
+        {
+            let keys =
+                non_inlined_keys.lock().unwrap().drain(..).map(|(_, key)| key).collect::<Vec<_>>();
+            let values_read = non_inlined_entries.clone();
+            StoreParallelIterator::lookup_keys(
+                store.clone(),
+                DBCol::State,
+                keys,
+                6,
+                move |key, value| {
+                    assert!(value.is_some(), "Key not found: {}", hex::encode(&key));
+                    let value = value.unwrap().to_vec();
+                    let mut values = values_read.lock().unwrap();
+                    values.insert(key.to_vec(), value);
+                },
+                true,
+            )
+        }
+
+        // Now that we've read all the non-inlined keys into memory, delete the State column, and
+        // write back these values.
+        let mut update = store.store_update();
+        update.delete_all(DBCol::State);
```

Comment:
so i wonder if it is worth adding a way to delete a column like this via drop_cf(), like we do for checkpoints, because otherwise the space wont be cleared up until a compaction happens at some point. obviously not a big deal to implement that in this PR, but in the future it would prob be good

Reply:
Aggressive trimming is just a testing tool for this trimming, it isn't meant to shrink the db size. But you're right, when it comes time to maybe actually delete most of the State column, then yeah, drop_cf would be awesome.